p2p/src/net/types/protocol.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2021-2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::{self, Display, Formatter};
+
+use common::primitives::semver::SemVer;
+use serialization::{Decode, Encode};
+
+/// Protocol type and version.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash, Encode, Decode)]
+pub struct Protocol {
```

Comment:
I think calling this name "Protocol" will be a ripe source of confusion. Perhaps "SupportedProtocol"? Though also supported protocol has its problems that I'll mention in another comment.

Reply:
I agree with you about the naming, but I don't like `SupportedProtocol` either. I think that the emphasis should be on what this protocol is or at least on its "level". I mean that we want to abstract libp2p "raw" p2p protocols and introduce our higher level protocol. I don't have better ideas than "application protocol" for now, that I also don't like.

Reply:
https://github.com/mintlayer/mintlayer-core/issues/371